src/Framework/IProjectElement.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-----------------------------------------------------------------------
+// </copyright>
+// <summary>Interface for exposing a ProjectElement to the appropriate loggers.</summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Interface for exposing a ProjectElement to the appropriate loggers
+    /// </summary>
+    public interface IProjectElement
+    {
+
+        /// <summary>
+        /// Gets the name of the associated element. 
+        /// Useful for display in some circumstances.
+        /// </summary>
+        string ElementName { get; }
+
+
+        /// <summary>
+        /// The outer xml markup of the xml element associated with this project element/>
+        /// </summary>
+        string OuterXmlElement { get; }
```

Comment:
Can you rename this to hide the fact that it's XML? Maybe `OuterElement` or `RawElementRepresentation`, or `RawText`

Reply:
Mmhh.. but this was originally being retrieved directly as this.XmlElement.OuterXml. So I thought it was important to communicate the original provenance. I would have exposed XmlElementWithLocation directly (the type of XmlElement), but that needed an extra project reference, and avoiding that was the main purpose of this interface.
But I'm not well versed in the content of a ProjectElement, so I'll follow your advice here.

Reply:
The ProjectElement wraps over an XmlElement. It is like an MSBuild AST node. I was thinking that if we ever change MSBuild's representation from XML to something else (probably never, but who knows), then having an interface member referencing XML would be cumbersome.

Reply:
Ok, makes sense, will change